### PR TITLE
Bump the max table size to 103

### PR DESCRIPTION
**Pull request by kochmaxence**
feature gate 'insane-tables'

Used "insane-tables" instead of "really-huge-tables" to express the fact that it is likely a bad idea.

see https://github.com/diesel-rs/diesel/issues/359#issuecomment-227920184

### CODE REVIEW

File: diesel/Cargo.toml [hunk 1]
```diff
@@ -45,6 +45,7 @@ unstable = []
 lint = ["clippy"]
 large-tables = []
 huge-tables = ["large-tables"]
+insane-tables = ["huge-tables"]
```
Comment (killercup) [inline diesel/Cargo.toml]: Can we go with "humongous-tables"? It fits better as the X in "large, huge, X"

**Comment by sgrif** [comment]
I'm also curious what the compile times are with this feature on. I think we can probably just update `huge-tables` to this number instead of adding a new feature

**Comment by kochmaxence** [comment]
It takes me about 15 minutes to compile disel with the feature.

Fifteen. Minutes.

image

I can cope with "humongous-tables", instead of "insane-tables", "probably-a-bad-idea-tables", "intern-tables", "unreadable-tables", "anti-compile-tables" & others.

Unfortunately I have a table with 80 columns.

**Comment by Eijebong** [comment]
Why are you putting 103 if your table only has 80 ?

**Comment by theduke** [comment]
15 minutes is a lot better then when I tried to implement the same thing.
It took around an hour for me with 128 columns, if I remember correctly.

It's still way too long to be sensible though, imo.

**Comment by leodasvacas** [comment]
We should not use "insane" in a name because that may be perceived as [ableist](https://en.wikipedia.org/wiki/Ableism).

**Comment by nsrsr** [comment]
no, we should not. Please let's all work together on software instead of polluting it with so called "social issues" and politics of any kind. Please spare my area of work from left-wing ideas. Thanks.

**Comment by sgrif** [comment]
@nsrsr Objecting to the term "insane" in a feature name in favor of another word is a completely reasonable request. It very much falls under "Using welcoming and inclusive language" [which we've laid out in this project's values very clearly](https://github.com/diesel-rs/diesel/blob/master/code_of_conduct.md).

Attacking people for such ideas will not be tolerated. This is your only warning.

**Comment by nsrsr** [comment]
I do not agree to this left-wing code of conduct if it forbids me to speak freely. I'm an advocate for free speech. This will be my last comment in this project, that's why I've written a few sentences. Thus whatever actions you choose to do now don't matter. 
I'm mentally ill myself (severely disabled, acknowledged by the government) and of course I'm insane. People are insane, some at least. This code of conduct is insane too. To hell with it. The people who created the code of conduct are political activists and insane too. Did they ask me if I don't want to be labeled with the truth? No! I know they're insane, because I was hospitalized many times and have seen all types of mental illnesses. I can even detect small traces of insanity really easy. I look at people and then they speak, yup, it's all settled, they can't be helped anymore, these people just walk around with you and me. And they're free to go! Shame! There are no medications for leftism! I've got my personal bullshit detector and you're really triggering me. 
I was hospitalized many times and I found loving people there which were broken and thus called insane, these people mostly just need a batch of love, right? But nobody is giving them love, just pills, and they're insane and non-functional, that's how cruel nature is. And I experienced, if they can't help themselves nobody can help them. Because they are severely broken. 
But then what I find daily going on in the real world really is insane. And we're now debating if a non-living piece of software may be called insane. Oh, how far have we come. Am I demented yet? Is this an illusion? No. This is marxism: everything is political. Destroying friendships and relationships, infecting everything, infecting my safe space. To say it with leftist words: programming is my safe space -- dare you infect it with politics, be it left or right! 
Now I am really insane and taking meds which keep me alive (yuck!) and you show what your code of conduct really is: politics. I should be able to voice my opinion (it's nothing more, nothing less, of course) that things and people can be insane and thus be called insane like myself I am, because that's what I am. And so you need to ostracize me, is that what your code of conduct is about? Ostracizing mentally ill people, because one time in history there's one who knows what he's talking about? Then better drop that document.

We need to be clear and direct if we voice our opinion so there are no misunderstandings. We should not complicate and castrate our languages in order to make people feel better. We should help people to help themselves. Only then they will feel better and live in dignity. No code of conduct will help, it help's to cement insanity.
Thanks for trying to hinder me speaking freely. You can thank god that I'm not really suicidal and thus triggered by your behavior to threaten me with consequences. Your code of conduct does not really matter. I'm not even questioning it anymore seeing how you want to expel different-minded people, I know the door. I wish you good luck with it. -- Oh god, people, why won't you wake up. 'cause if one can't take pills nowadays, one's really lost.

**Comment by kochmaxence** [comment]
I'm closing this for several reasons:

1. Bumping the allowed table size is a bad idea that originates from a bad idea, which will cause more drama in the long term: issues with "I need to support 150 columns". Large tables are a bad design choice.
2. The increased compile time is subject to drama in issues.
3. The drama around how to name things is driving me insane (pun intended).
4. I opened this in Oct. and kept it open until now because I still don't feel like it's the correct approach.

-> 5. If anyone needs to arbitrarily increase the maximum size, it's better to let them fork & modify to their needs. Maybe we should just document it somewhere, I might just open a new PR with some documentation and snippets I used to auto-generate it.

That said, I feel like @nsrsr had a point (in his deleted comment) about the issues we bring into our open-source community from society. But it's not the place to discuss this.

**Pull request closed without merging**